### PR TITLE
Improve GH action to produce release artifacts

### PR DESCRIPTION
**Pull request by dalg24**
* Generate the `kokkos-4.X.YZ-SHA-256.txt` hashes
* Consolidate upload and download steps

The maintainer is only left with certifying the cryptographic hashes.
```
curl -sL https://github.com/kokkos/kokkos/releases/download/4.X.YZ/kokkos-4.X.YZ.zip -o kokkos-4.X.YZ.zip
curl -sL https://github.com/kokkos/kokkos/releases/download/4.X.YZ/kokkos-4.X.YZ.tar.gz -o kokkos-4.X.YZ.tar.gz
curl -sL https://github.com/kokkos/kokkos/releases/download/4.X.YZ/kokkos-4.X.YZ-SHA-256.txt -o kokkos-4.X.YZ-SHA-256.txt
curl -sL https://github.com/kokkos/kokkos/releases/download/4.X.YZ/kokkos-4.X.YZ.intoto.jsonl -o kokkos-4.X.YZ.intoto.jsonl
grep kokkos-4.X.YZ.tar.gz kokkos-4.X.YZ-SHA-256.txt | shasum -c
grep kokkos-4.X.YZ.zip kokkos-4.X.YZ-SHA-256.txt | shasum -c
slsa-verifier verify-artifact kokkos-4.X.YZ.tar.gz --provenance-path kokkos-4.X.YZ.intoto.jsonl --source-uri github.com/kokkos/kokkos --source-tag 4.X.YZ
slsa-verifier verify-artifact kokkos-4.X.YZ.zip --provenance-path kokkos-4.X.YZ.intoto.jsonl --source-uri github.com/kokkos/kokkos --source-tag 4.X.YZ
```
Then unarchive both the zip and tar.gz and diff it with your local copy before certifying and uploading the signature.

### CODE REVIEW

**Comment by dalg24** [comment]
(in case anyone cares, tested here https://github.com/dalg24/kokkos/releases/tag/5.0.02)

File: .github/workflows/releases.yml [hunk 1]
```diff
@@ -16,30 +16,24 @@ jobs:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
       - name: Build artifacts
         run: |
-          git archive -o kokkos-${{ github.ref_name }}.zip HEAD
-          git archive -o kokkos-${{ github.ref_name }}.tar.gz HEAD
+          git archive --prefix=kokkos-${{ github.ref_name }}/ -o kokkos-${{ github.ref_name }}.zip HEAD
+          git archive --prefix=kokkos-${{ github.ref_name }}/ -o kokkos-${{ github.ref_name }}.tar.gz HEAD
```
Comment (dalg24) [inline .github/workflows/releases.yml:20]: Fixes #7232

**Review by lucbv — CHANGES_REQUESTED** [review]
See my comment to remove a line when generating the sha256s, also why do you use grep in the verification step? I believe if you just do `shasum -c mycheckfile-SHA256.txt` will work just fine?

File: .github/workflows/releases.yml [hunk 2]
```diff
@@ -16,30 +16,24 @@ jobs:
...
 
       - name: Generate hashes
         shell: bash
         id: hash
         run: |
           # sha256sum generates sha256 hash for all artifacts.
           # base64 -w0 encodes to base64 and outputs on a single line.
-          echo "hashes=$(sha256sum kokkos-${{ github.ref_name }}.zip kokkos-${{ github.ref_name }}.tar.gz | base64 -w0)" >> "$GITHUB_OUTPUT"
+          sha256sum kokkos-${{ github.ref_name }}.zip > kokkos-${{ github.ref_name }}-SHA-256.txt
```
[the remark this answers is not in the document]
Comment (dalg24) [reply .github/workflows/releases.yml]: Done

File: .github/workflows/releases.yml [hunk 3]
```diff
@@ -16,30 +16,23 @@ jobs:
...
+          git archive --prefix=kokkos-${{ github.ref_name }}/ -o kokkos-${{ github.ref_name }}.tar.gz HEAD
 
       - name: Generate hashes
         shell: bash
         id: hash
         run: |
           # sha256sum generates sha256 hash for all artifacts.
           # base64 -w0 encodes to base64 and outputs on a single line.
-          echo "hashes=$(sha256sum kokkos-${{ github.ref_name }}.zip kokkos-${{ github.ref_name }}.tar.gz | base64 -w0)" >> "$GITHUB_OUTPUT"
+          sha256sum kokkos-${{ github.ref_name }}.zip  kokkos-${{ github.ref_name }}.tar.gz > kokkos-${{ github.ref_name }}-SHA-256.txt
```
Comment (masterleinad) [inline .github/workflows/releases.yml:28]:
```suggestion
          sha256sum kokkos-${{ github.ref_name }}.zip kokkos-${{ github.ref_name }}.tar.gz > kokkos-${{ github.ref_name }}-SHA-256.txt
```

**Review by masterleinad — APPROVED** [review]

File: .github/workflows/releases.yml (hunk 3 above)
Comment (dalg24) [reply .github/workflows/releases.yml]: Thanks.  Amended to the last commit

**Comment by lucbv** [comment]
I see your point, I guess in our case I assumed we want know if one of them is not downloaded for some reason but for users it makes sense to check individually.

**Comment by dalg24** [comment]
> I see your point, I guess in our case I assumed we want know if one of them is not downloaded for some reason but for users it makes sense to check individually.

Didn't realize you were referring to maintainer instructions I wrote in the description and I agree with you in that case `grep` is unnecessary.

**Comment by dalg24** [comment]
NVC++ segfault and OpenMPTarget runtime failures are clearly unrelated.

**Pull request merged**